waku/v2/node/peer_manager/peer_manager.nim

```diff
@@ -477,31 +477,41 @@ proc connectToNodes*(pm: PeerManager,
   # later.
   await sleepAsync(chronos.seconds(5))
 
-# Returns the amount of physical connections for a given direction
+# Returns the amount of physical connections (in and out)
 # containing at least one stream with the given protocol.
-proc getNumConnections*(pm: PeerManager, dir: Direction, protocol: string): int =
-  var numConns = 0
+proc getNumConnections*(pm: PeerManager, protocol: string): (int, int) =
+  var
+    numConnsIn = 0
+    numConnsOut = 0
   for peerId, muxers in pm.switch.connManager.getConnections():
     for peerConn in muxers:
       let streams = peerConn.getStreams()
-      if peerConn.connection.transportDir == dir:
-        if streams.anyIt(it.protocol == protocol):
-          numConns += 1
-  return numConns
-
-proc getNumStreams*(pm: PeerManager, dir: Direction, protocol: string): int =
-  var numConns = 0
+      if streams.anyIt(it.protocol == protocol):
+        if peerConn.connection.transportDir == Direction.In:
```

Comment:
perhaps nitpick but can be have "if In" and "else if Out"? Just in case there is an uninitialized value or something weird coming from libp2p.

if peerConn.connection.transportDir == Direction.In:
  xxx
else if peerConn.connection.transportDir == Direction.Out:
  xxx

Reply:
Yeah, absolutely, I was originally gonna do that, but then was not sure if it can be uninitialized (cause it did not seem like being a refernce), but maybe I misunderstood how that works in Nim:)